app/src/main/java/com/cornellappdev/android/eateryblue/ui/viewmodels/UpcomingViewModel.kt

```diff
@@ -14,66 +16,79 @@ import kotlinx.coroutines.flow.asStateFlow
-            is EateryApiResponse.Pending -> EateryApiResponse.Pending
-            is EateryApiResponse.Success -> {
-                EateryApiResponse.Success(
-                    apiResponse.data.filter {
-                        passesFilter(it, filters)
-                    })
+    val eateryFlow =
+        eateryRepository.eateryFlow.combine(_locationFilterFlow) { apiResponse, filters ->
+            when (apiResponse) {
+                is EateryApiResponse.Error -> EateryApiResponse.Error
+                is EateryApiResponse.Pending -> EateryApiResponse.Pending
+                is EateryApiResponse.Success -> {
+                    EateryApiResponse.Success(
+                        apiResponse.data.filter {
+                            passesFilter(it, filters)
+                        })
+                }
             }
-        }
-    }.stateIn(viewModelScope, SharingStarted.Eagerly, EateryApiResponse.Pending)
+        }.stateIn(viewModelScope, SharingStarted.Eagerly, EateryApiResponse.Pending)
 
-    fun addFilter(filter: Filter) = viewModelScope.launch {
+    fun addLocationFilter(filter: Filter) = viewModelScope.launch {
         addLocationFilters(filter)
     }
 
-    fun removeFilter(filter: Filter) = viewModelScope.launch {
-        val newList = _filtersFlow.value.toMutableList()
+    fun removeLocationFilter(filter: Filter) = viewModelScope.launch {
+        val newList = _locationFilterFlow.value.toMutableList()
         newList.remove(filter)
-        _filtersFlow.value = newList
+        _locationFilterFlow.value = newList
     }
 
-    fun addMealFilters(filters: List<Filter>) = viewModelScope.launch {
-        val newList = _filtersFlow.value.toMutableList()
-        newList.removeAll(Filter.MEALS)
-        newList.addAll(filters)
-        _filtersFlow.value = newList
+    fun changeMealFilter(filter: MealFilter) = viewModelScope.launch {
+        _mealFilterFlow.value = filter
     }
 
-    fun resetFilters() = viewModelScope.launch {
-        _filtersFlow.value = listOf()
+    fun resetLocationFilters() = viewModelScope.launch {
+        _locationFilterFlow.value = listOf()
     }
 
     private fun addLocationFilters(filter: Filter) = viewModelScope.launch {
-        val newList = _filtersFlow.value.toMutableList()
+        val newList = _locationFilterFlow.value.toMutableList()
         newList.add(filter)
         val locations = newList.filter { Filter.LOCATIONS.contains(it) }
         if (locations.size == 3) {
             newList.removeAll(Filter.LOCATIONS)
         }
-        _filtersFlow.value = newList
+        _locationFilterFlow.value = newList
+    }
+
+    private fun nextMeal(): MealFilter? {
+        Log.d(
+            "Current Time",
+            (LocalDateTime.now().hour + LocalDateTime.now().minute / 60f).toString()
+        )
+        return MealFilter.values()
+            .find { it.endTimes >= LocalDateTime.now().hour + LocalDateTime.now().minute / 60f }
     }
```

Comment:
ok i know the rest of this file aint got it but like

Please write a spec for this function, even if it's private!